scalr/data_source_scalr_agent_pool.go

```diff
@@ -54,6 +60,7 @@ func dataSourceScalrAgentPoolRead(ctx context.Context, d *schema.ResourceData, m
 
 	agentPoolID := d.Get("id").(string)
 	name := d.Get("name").(string)
+	vcsEnabled := d.Get("vcs_enabled").(bool)
```

Comment:
This evaluates to `false` if `vcs_enabled` is not set in configuration. Should test the value with `d.GetOk()`.

Reply:
According to docs default value will be returned which is desired behaviour.
```
If the key does exist in the schema but doesn't exist in the configuration,
then the default value for that type will be returned. For strings, this is
"", for numbers it is 0, etc.
```

Reply:
PS: In provider code we have a lot similar cases like:
scalr.Bool(d.Get("export_shell_variables").(bool))

Reply:
@lyzohub 
> According to docs default value will be returned which is desired behaviour.

This is desired behaviour in general, but not in this case. There is no way to search for the pool without `vcs_enabled` filter.
Given that I don't know/care about this flag, the following snippet
```
data "scalr_agent_pool" "default" {
  name       = "default-pool"
  account_id = "acc-xxxxxxx"
}
```
will perform the search with `vcs_enabled=false` filter.

Reply:
@petroprotsakh adjusted logic, now it works as expected